types/lib/users.d.ts

```diff
@@ -67,17 +145,29 @@ export interface DeletePhoneNumberResponse extends BaseResponse {
 export interface DeleteWebAuthnRegistrationResponse extends BaseResponse {
     user_id: UserID;
 }
+export declare class UserSearchIterator {
+    private client;
+    private data;
+    private mode;
+    constructor(client: Users, data: SearchRequest);
+    next(): Promise<User[]>;
+    hasNext(): boolean;
+    [Symbol.asyncIterator](): AsyncIterator<User[]>;
+}
 export declare class Users {
     base_path: string;
     private client;
     constructor(client: AxiosInstance);
     private endpoint;
     create(data: CreateRequest): Promise<CreateResponse>;
     get(userID: UserID): Promise<GetResponse>;
+    search(data: SearchRequest): Promise<SearchResponse>;
+    searchAll(data: SearchRequest): UserSearchIterator;
```

Comment:
For reading clarity, should we name this `searchAll`, or `getAll`? The intended purpose is to "get all users", correct? I suppose it isn't written explicitly to do get everything, just make iterating easier.

Reply:
the intended purpose is to handle paging through all results of a search- so.... sometimes?
 I feel like `getAll` would be better as its own helper method, eg `() => searchAll({ /* empty set of filters */ })`

Reply:
That makes sense to me.